### ISSUE TITLE
[6.0.1] Form submission fails every time if it fails once.

### ISSUE THREAD

**Issue by malreynolds** [erikras/redux-form#1670]
Hi, if I submit a form and the asynchronous validation fails, the form doesn't want to submit again, regardless of whether I change the form fields. 

In my use case, I have a `Login` component, which calls `login()` on `onSubmit`. 

This is what `login()` looks like.

```javascript
export function login({ email, password, dispatch }) {
    return fetch('/rest-auth/login/',{
        method: 'POST',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({
            email: email,
            password: password,
        })
    }).then(response => {
        if (response.status == 200) {
            dispatch({type: AUTH_USER});
            return response.json().then(data => {
                localStorage.setItem('token', data.token);
                browserHistory.push('/navigation');
            })
        }
        else return response.json().then(errors => {
            throw new SubmissionError({...errors, _error: 'Login failed. ' +
                                       (errors['non_field_errors'] ? errors['non_field_errors'][0] : '')})
        })
    })
}
```

The error fields are being correctly set as my API response fields have the same names as my form fields.

The issue is that when I try to resubmit the form with the fixed fields, it immediately fails, in fact it doesn't even get to the login function. I added some debug code in the shipped handleSubmit and it looks like there is a `if (valid)` and valid is always false. 

The same code works with 6.0.0-rc.4 and breaks after upgrade to 6.0.1.

**Comment by ortex** [comment]
It was fixed but not yet released. #1641

**Comment by aumayr** [comment]
Released and fixed in [`v6.0.2`](https://github.com/erikras/redux-form/releases/tag/v6.0.2)  🎉

**Comment by malreynolds** [comment]
Tested with the new version, issue is now fixed.
